### PR TITLE
CRM457-1557: Add search UI

### PR DESCRIPTION
**Pull request by patrick-laa**
## Description of change

[Link to relevant ticket](https://dsdmoj.atlassian.net/browse/CRM457-1557)

## Screenshots of changes:
Screenshot 2024-07-08 at 14 54 52

## How to manually test the feature
... Wait for a back-end or hack the AppStoreClient to return a payload similar to the specs.

### CODE REVIEW

**Review by ivanELEC — APPROVED** [review]

File: app/controllers/prior_authority/searches_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module PriorAuthority
+  class SearchesController < BaseController
+    def show
+      @search = Search.new(search_params)
```
Comment (dwhenry) [inline app/controllers/prior_authority/searches_controller.rb:4]: so every other one of these is called `SearchForm`.. whould we do the same here?

File: app/models/search_result.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class SearchResult
```
Comment (dwhenry) [inline app/models/search_result.rb:1]: should this live under `view_models` as it is a model built from the JSON response?

File: app/models/search_result.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  def submission
+    @submission ||= Submission.find_by(id: @data[:application_id]) || UpdateSubmission.call(@data.deep_stringify_keys)
+  end
+
+  def laa_reference
+    submission.data['laa_reference']
+  end
+
+  def firm_name
+    submission.data['firm_name']
```
Comment (dwhenry) [inline app/models/search_result.rb:20]: any reason we can't look this up form the response JSON? try and avoid touching the local DB as much as possible?

**Review by dwhenry — APPROVED** [review]
I like it

**Pull request merged**